retrievalmarket/impl/clientstates/client_states.go

```diff
@@ -20,7 +20,12 @@ type ClientDealEnvironment interface {
 
 // SetupPaymentChannel sets up a payment channel for a deal
 func SetupPaymentChannel(ctx fsm.Context, environment ClientDealEnvironment, deal rm.ClientDealState) error {
-	paych, err := environment.Node().GetOrCreatePaymentChannel(ctx.Context(), deal.ClientWallet, deal.MinerWallet, deal.TotalFunds)
+	tok, _, err := environment.Node().GetChainHead(ctx.Context())
+	if err != nil {
+		return ctx.Trigger(rm.ClientEventPaymentChannelErrored, err)
```

Comment:
@hannahhoward Should I create a new error for this state - or will the one which I've used suffice?

Reply:
I think it's fine

Reply:
okie dokie